Index: scalapack/CMakeLists.txt
===================================================================
--- scalapack.orig/CMakeLists.txt	2023-10-25 17:15:50.173452752 +0200
+++ scalapack/CMakeLists.txt	2023-10-25 17:15:50.165452685 +0200
@@ -234,6 +234,7 @@
 
 if (UNIX)
    add_library(scalapack ${blacs} ${tools} ${tools-C} ${extra_lapack} ${pblas} ${pblas-F} ${ptzblas} ${ptools} ${pbblas} ${redist} ${src} ${src-C})
+   set_target_properties(scalapack PROPERTIES OUTPUT_NAME scalapack-${MPI_FLAVOR} )
    set_target_properties(scalapack PROPERTIES
     VERSION ${SCALAPACK_VERSION}
     SOVERSION ${SCALAPACK_VERSION_MAJOR}.${SCALAPACK_VERSION_MINOR})
@@ -306,9 +307,9 @@
 configure_file(${SCALAPACK_SOURCE_DIR}/CMAKE/scalapack-config-build.cmake.in
   ${SCALAPACK_BINARY_DIR}/scalapack-config.cmake @ONLY)
 
-configure_file(${CMAKE_CURRENT_SOURCE_DIR}/scalapack.pc.in ${CMAKE_CURRENT_BINARY_DIR}/scalapack.pc)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/scalapack.pc.in ${CMAKE_CURRENT_BINARY_DIR}/scalapack-${MPI_FLAVOR}.pc)
  install(FILES
-  ${CMAKE_CURRENT_BINARY_DIR}/scalapack.pc
+  ${CMAKE_CURRENT_BINARY_DIR}/scalapack-${MPI_FLAVOR}.pc
   DESTINATION ${PKG_CONFIG_DIR}
    )
 
@@ -317,8 +318,8 @@
 install(FILES
   ${SCALAPACK_BINARY_DIR}/CMakeFiles/scalapack-config.cmake
   ${SCALAPACK_BINARY_DIR}/scalapack-config-version.cmake
-  DESTINATION lib/cmake/scalapack-${SCALAPACK_VERSION}
+  DESTINATION lib/cmake/scalapack-${SCALAPACK_VERSION}.${MPI_FLAVOR}
   )
 
 install(EXPORT scalapack-targets
-  DESTINATION lib/cmake/scalapack-${SCALAPACK_VERSION})
+  DESTINATION lib/cmake/scalapack-${SCALAPACK_VERSION}.${MPI_FLAVOR})
Index: scalapack/scalapack.pc.in
===================================================================
--- scalapack.orig/scalapack.pc.in	2022-02-16 13:01:18.474865257 +0100
+++ scalapack/scalapack.pc.in	2023-10-25 17:17:27.610273570 +0200
@@ -5,5 +5,5 @@
 Description: SCALAPACK reference implementation
 Version: @SCALAPACK_VERSION@
 URL: http://www.netlib.org/scalapack/
-Libs: -L${libdir} -lscalapack
+Libs: -L${libdir} -lscalapack-@MPI_FLAVOR@
 Requires: mpi lapack blas
