Index: scalapack-2.0.2/CMakeLists.txt
===================================================================
--- scalapack-2.0.2.orig/CMakeLists.txt
+++ scalapack-2.0.2/CMakeLists.txt
@@ -7,6 +7,11 @@ configure_file(
   COPYONLY
 )
 
+set(SCALAPACK_MAJOR_VERSION 2)
+set(SCALAPACK_MINOR_VERSION 1)
+set(SCALAPACK_PATCH_VERSION 0)
+set(SCALAPACK_VERSION "${SCALAPACK_MAJOR_VERSION}.${SCALAPACK_MINOR_VERSION}.${SCALAPACK_PATCH_VERSION}")
+
 # Add the CMake directory for custon CMake modules
 set(CMAKE_MODULE_PATH "${SCALAPACK_SOURCE_DIR}/CMAKE" ${CMAKE_MODULE_PATH})
 
@@ -231,14 +236,32 @@ append_subdir_files(src "SRC")
 append_subdir_files(src-C "SRC")
 
 if (UNIX)
+   MESSAGE(STATUS "Building only static")
+   if(BUILD_SHARED_LIBS)
+   add_library(scalapack SHARED ${blacs} ${tools} ${tools-C} ${extra_lapack} ${pblas} ${pblas-F} ${ptzblas} ${ptools} ${pbblas} ${redist} ${src} ${src-C})
+   set_target_properties(scalapack PROPERTIES OUTPUT_NAME scalapack-${MPI_FLAVOR} )
+   set_target_properties(scalapack PROPERTIES
+    VERSION ${SCALAPACK_MAJOR_VERSION}.${SCALAPACK_MINOR_VERSION}.${SCALAPACK_PATCH_VERSION}
+    SOVERSION ${SCALAPACK_MAJOR_VERSION}.${SCALAPACK_MINOR_VERSION})
+   else()
    add_library(scalapack ${blacs} ${tools} ${tools-C} ${extra_lapack} ${pblas} ${pblas-F} ${ptzblas} ${ptools} ${pbblas} ${redist} ${src} ${src-C})
+   endif()
    target_link_libraries( scalapack ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES})
    scalapack_install_library(scalapack)
 else (UNIX) # Need to separate Fortran and C Code
+   MESSAGE(STATUS "Building shared")
    OPTION(BUILD_SHARED_LIBS "Build shared libraries" ON )
-   add_library(scalapack ${blacs} ${tools-C} ${pblas} ${ptools} ${redist} ${src-C})
+   add_library(scalapack SHARED ${blacs} ${tools-C} ${pblas} ${ptools} ${redist} ${src-C})
+   set_target_properties(scalapack PROPERTIES OUTPUT_NAME scalapack-${MPI_FLAVOR})
+   set_target_properties(scalapack PROPERTIES
+    VERSION ${SCALAPACK_MAJOR_VERSION}.${SCALAPACK_MINOR_VERSION}.${SCALAPACK_PATCH_VERSION}
+    SOVERSION ${SCALAPACK_MAJOR_VERSION}.${SCALAPACK_MINOR_VERSION})
    target_link_libraries( scalapack ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES})
-   add_library(scalapack-F ${pblas-F} ${pbblas} ${ptzblas} ${tools} ${src} ${extra_lapack} )
+   add_library(scalapack-F SHARED ${pblas-F} ${pbblas} ${ptzblas} ${tools} ${src} ${extra_lapack} )
+   set_target_properties(scalapack-F PROPERTIES OUTPUT_NAME scalapack-F-${MPI_FLAVOR})
+   set_target_properties(scalapack-F PROPERTIES
+    VERSION ${SCALAPACK_MAJOR_VERSION}.${SCALAPACK_MINOR_VERSION}.${SCALAPACK_PATCH_VERSION}
+    SOVERSION ${SCALAPACK_MAJOR_VERSION}.${SCALAPACK_MINOR_VERSION})
    target_link_libraries( scalapack-F ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES})
    scalapack_install_library(scalapack)
    scalapack_install_library(scalapack-F)
@@ -251,10 +274,10 @@ add_subdirectory(TESTING)
 SET(CPACK_PACKAGE_NAME "ScaLAPACK")
 SET(CPACK_PACKAGE_VENDOR "University of Tennessee, Univ. of California Berkeley, Univ. of Colorado Denver and NAG Ltd")
 SET(CPACK_PACKAGE_DESCRIPTION_SUMMARY "ScaLAPACK- Linear Algebra Package")
-set(SCALAPACK_VERSION 2.1.0)
-set(CPACK_PACKAGE_VERSION_MAJOR 2)
-set(CPACK_PACKAGE_VERSION_MINOR 1)
-set(CPACK_PACKAGE_VERSION_PATCH 0)
+#set(SCALAPACK_VERSION 2.1.0)
+set(CPACK_PACKAGE_VERSION_MAJOR ${SCALAPACK_MAJOR_VERSION})
+set(CPACK_PACKAGE_VERSION_MINOR ${SCALAPACK_MINOR_VERSION})
+set(CPACK_PACKAGE_VERSION_PATCH ${SCALAPACK_PATCH_VERSION})
 set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_CURRENT_SOURCE_DIR}/LICENSE")
 SET(CPACK_PACKAGE_INSTALL_DIRECTORY "SCALAPACK")
 IF(WIN32 AND NOT UNIX)
